core/src/main/java/nl/nn/adapterframework/compression/ZipWriter.java

```diff
@@ -98,6 +98,7 @@ public MessageZipEntry createPart(Message input, PipeLineSession session, Parame
 			if (StringUtils.isEmpty(filename) && contents != input) {
 				filename = input.asString();
 			}
+			session.unscheduleCloseOnSessionExit(contents);
```

Comment:
This line is the actual fix.

It could also be called in the Collection class when adding to the collection, or in the CollectorPipeBase.

Not sure what really is best, I decided to add this here because it's the ZipWriter which actually uses the files and does not consume the contents until the time of building. Another Collector implementation might consume the files early and then this is not needed.